compaction.go

```diff
@@ -1,16 +1,55 @@
 package lsm
 
 func (db *DB) compactLevel(level int) error {
-	if level == 0 {
+	if level == 0 && db.needL0Compaction() {
 		return db.compactLevel0()
+	} else if level > 0 && db.needLevelNCompaction(level) {
+		return db.compactLevelN(level)
 	}
-	return db.compactLevelN(level)
+	return nil
 }
 
 func (db *DB) compactLevel0() error {
+	iterators0 := make([]*SSTableIterator, 0)
+	for _, sstable := range db.levels[0].sstables {
+		iterators0 = append(iterators0, sstable.Iterator())
+	}
+	iterators1 := make([]*SSTableIterator, 0)
+	for _, sstable := range db.levels[1].sstables {
+		iterators1 = append(iterators1, sstable.Iterator())
+	}
+
+	iterators := append(iterators0, iterators1...)
+	sstList, err := db.mergeIterators(iterators, 1)
+	if err != nil {
+		return err
+	}
+	db.deleteSStableAtLevel(0, iterators0)
```

Comment:
### Security Issues
Ensure that the `db.deleteSStableAtLevel(level, iterators)` method securely deletes the SSTables. Use `os.RemoveAll` instead of `os.Remove` to ensure that all files and directories are securely deleted. Additionally, consider adding a check to ensure that the files being deleted are indeed SSTables and not arbitrary files.